apps/spruce/src/pages/version/TestAnalysis/index.tsx

```diff
@@ -0,0 +1,215 @@
+import { useState } from "react";
+import { useQuery } from "@apollo/client";
+import styled from "@emotion/styled";
+import Button from "@leafygreen-ui/button";
+import { Combobox, ComboboxOption } from "@leafygreen-ui/combobox";
+import { BasicEmptyState } from "@leafygreen-ui/empty-state";
+import { palette } from "@leafygreen-ui/palette";
+import { SearchInput } from "@leafygreen-ui/search-input";
+import { ListSkeleton } from "@leafygreen-ui/skeleton-loader";
+import { H3, Label, Body, H3Props } from "@leafygreen-ui/typography";
+import pluralize from "pluralize";
+import { failedTaskStatuses, taskStatusToCopy } from "constants/task";
+import { size } from "constants/tokens";
+import {
+  TestAnalysisQuery,
+  TestAnalysisQueryVariables,
+} from "gql/generated/types";
+import { TEST_ANALYSIS } from "gql/queries";
+import FailedTestGroup from "./FailedTestGroup";
+import {
+  countTotalTests,
+  filterGroupedTests,
+  getAllBuildVariants,
+  getAllTaskStatuses,
+  groupTestsByName,
+} from "./utils";
+
+const { green } = palette;
+interface TestAnalysisProps {
+  versionId: string;
+}
+const TestAnalysis: React.FC<TestAnalysisProps> = ({ versionId }) => {
+  const [selectedTaskStatuses, setSelectedTaskStatuses] = useState<string[]>(
+    [],
+  );
+  const [selectedBuildVariants, setSelectedBuildVariants] = useState<string[]>(
+    [],
+  );
+  const [searchValue, setSearchValue] = useState<string>("");
+  const { data, loading } = useQuery<
+    TestAnalysisQuery,
+    TestAnalysisQueryVariables
+  >(TEST_ANALYSIS, {
+    variables: {
+      versionId,
+      options: {
+        statuses: failedTaskStatuses,
+      },
+      opts: {
+        statuses: ["fail"],
+      },
+    },
+  });
+
+  const groupedTestsMap = groupTestsByName(
+    data ? data?.version?.tasks?.data : [],
+  );
+
+  const buildVariants = getAllBuildVariants(groupedTestsMap);
+  const taskStatuses = getAllTaskStatuses(groupedTestsMap);
+
+  const filteredGroupedTestsMap = filterGroupedTests(
+    groupedTestsMap,
+    searchValue,
+    selectedTaskStatuses,
+    selectedBuildVariants,
+  );
+  const groupedTestsMapEntries = Array.from(
+    filteredGroupedTestsMap.entries(),
+  ).sort((a, b) => b[1].length - a[1].length);
+
+  const numberOfTestsThatFailedOnMoreThanOneTask =
+    groupedTestsMapEntries.filter(([, tasks]) => tasks.length > 1).length;
+
+  const totalTestCount = countTotalTests(groupedTestsMap);
+  const totalFilteredTestCount = countTotalTests(filteredGroupedTestsMap);
+  const hasFilters =
+    selectedTaskStatuses.length > 0 ||
+    selectedBuildVariants.length > 0 ||
+    searchValue.length > 0;
+
+  const hasMatchingResults = totalFilteredTestCount > 0;
+  const hasResults = data && totalTestCount > 0;
+  return (
+    <div>
+      {loading ? (
+        <ListSkeleton />
+      ) : (
+        <div>
```

Comment:
[nit] Can use a fragment here since there's already a parent div

```suggestion
        <>
```